pyartifactory/objects.py

```diff
@@ -1,684 +1,745 @@
+    User,
+    Permission,
+    SimplePermission,
+    ArtifactPropertiesResponse,
+    ArtifactStatsResponse,
+)
+
+
+class Artifactory:
+    def __init__(
+        self,
+        url: str,
+        auth: Tuple[str, str] = None,
+        verify: bool = True,
+        cert: str = None,
+    ):
+        self.artifactory = AuthModel(url=url, auth=auth, verify=verify, cert=cert)
+        self.users = ArtifactoryUser(self.artifactory)
+        self.groups = ArtifactoryGroup(self.artifactory)
+        self.security = ArtifactorySecurity(self.artifactory)
+        self.repositories = ArtifactoryRepository(self.artifactory)
+        self.artifacts = ArtifactoryArtifact(self.artifactory)
+
+
+class ArtifactoryObject:
+    def __init__(self, artifactory: AuthModel) -> None:
+        self._artifactory = artifactory
+        self._auth = (
+            self._artifactory.auth[0],
+            self._artifactory.auth[1].get_secret_value(),
+        )
+        self._verify = self._artifactory.verify
+        self._cert = self._artifactory.cert
+        self.session = requests.Session()
+
+    def _get(self, route: str, **kwargs) -> Response:
+        """
+        :param route: API Route
+        :param kwargs: Additional parameters to add the request
+        :returns  An HTTP response
+        """
+        return self._generic_http_method_request("get", route, **kwargs)
+
+    def _post(self, route: str, **kwargs) -> Response:
+        """
+        :param route: API Route
+        :param kwargs: Additional parameters to add the request
+        :returns  An HTTP response
+        """
+        return self._generic_http_method_request("post", route, **kwargs)
+
+    def _put(self, route: str, **kwargs) -> Response:
+        """
+        :param route: API Route
+        :param kwargs: Additional parameters to add the request
+        :returns  An HTTP response
+        """
+        return self._generic_http_method_request("put", route, **kwargs)
+
+    def _delete(self, route: str, **kwargs) -> Response:
+        """
+        :param route: API Route
+        :param kwargs: Additional parameters to add the request
+        :returns  An HTTP response
+        """
+        return self._generic_http_method_request("delete", route, **kwargs)
+
+    def _generic_http_method_request(
+        self, method: str, route: str, **kwargs
+    ) -> Response:
+        """
+        :param method: HTTP method to use
+        :param route: API Route
+        :param kwargs: Additional parameters to add the request
+        :return: An HTTP response
+        """
+        raise_for_status = kwargs.pop("raise_for_status", True)
+        http_method = getattr(self.session, method)
+        response = http_method(
+            f"{self._artifactory.url}/{route}",
+            auth=self._auth,
+            **kwargs,
+            verify=self._verify,
+            cert=self._cert,
+        )
+        if raise_for_status:
+            response.raise_for_status()
+        return response
```

Comment:
Regarding previous comment, you can replace this section by just **response.raise_for_status()**

Reply:
yep, see previous comment for my rationale, and let me know if you want this seperated to a new PR.